packages/data-context/src/sources/ProjectDataSource.ts

```diff
@@ -210,6 +211,57 @@ export class ProjectDataSource {
     this._specWatcher.on('unlink', onSpecsChanged)
   }
 
+  private formatGlob (s: string, fallback: string) {
+    let [cleanedVal] = s.replace('{', '').replace('}', '').split(',')
+
+    if (cleanedVal?.includes('*')) {
+      cleanedVal = cleanedVal.replace(/\*/g, fallback)
+    }
+
+    return cleanedVal
+  }
+
+  async defaultSpecFileName () {
+    const defaultFileName = 'cypress/e2e/filename.cy.js'
+
+    try {
+      if (!this.ctx.currentProject || !this.ctx.coreData.currentTestingType) {
+        return null
+      }
+
+      let specPatternSet: string | undefined
+      const { specPattern = [] } = await this.ctx.project.specPatternsForTestingType(this.ctx.currentProject, this.ctx.coreData.currentTestingType)
+
+      if (Array.isArray(specPattern)) {
+        specPatternSet = specPattern[0]
+      }
+
+      if (!specPatternSet) {
+        return defaultFileName
+      }
+
+      const parsedGlob = parseGlob(specPatternSet)
+
+      let dirname = parsedGlob.path.dirname
+
+      if (dirname.startsWith('**')) {
+        dirname = dirname.replace('**', 'cypress')
+      }
+
+      const splittedDirname = dirname.split('/').filter((s) => s !== '**').map((x) => this.formatGlob(x, 'e2e')).join('/')
```

Comment:
Do we need to use `path.sep` here? I wonder if this will be a problem for windows.

Reply:
Right yes, I think this is correct, we always expect `/` in config files.